Modix.Services/Core/UserService.cs

```diff
@@ -92,18 +92,19 @@ public class UserService : IUserService
         /// <inheritdoc />
         public async Task<IUser> GetUserAsync(ulong userId)
         {
+            var restUser = await DiscordRestClient.GetUserAsync(userId);
             var user = (AuthorizationService.CurrentGuildId == null)
                 ? await DiscordClient.GetUserAsync(userId)
                 : await (await DiscordClient.GetGuildAsync(AuthorizationService.CurrentGuildId.Value))
                     .GetUserAsync(userId);
 
-            if (user == null)
+            if (user == null && restUser == null)
                 throw new InvalidOperationException($"Discord user {userId} does not exist");
 
             if (user is IGuildUser guildUser)
                 await TrackUserAsync(guildUser);
 
-            return user;
+            return user ?? restUser;
```

Comment:
So we're always pulling the REST user even though we might never do anything with it? :weird: